drivers/hmis/spec/requests/hmis/ce/project_ce_opportunities_spec.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+require 'rails_helper'
+require_relative '../login_and_permissions'
+require_relative './ce_spec_helper'
+
+RSpec.describe Hmis::GraphqlController, type: :request do
+  include_context 'ce spec helper'
+
+  describe 'project ceOpportunities query' do
+    let(:query) do
+      <<~GRAPHQL
+        query GetProjectCeOpportunities($id: ID!, $limit: Int = 25, $offset: Int = 0, $filters: CeOpportunityFilterOptions = null) {
+          project(id: $id) {
+            id
+            ceOpportunities(limit: $limit, offset: $offset, filters: $filters) {
+              offset
+              limit
+              nodesCount
+              nodes {
+                id
+                name
+                categories
+              }
+            }
+          }
+        }
+      GRAPHQL
+    end
+
+    let(:variables) do
+      {
+        id: project.id,
+      }
+    end
+
+    it "returns the project's opportunities" do
+      response, result = post_graphql(**variables) { query }
+      expect(response.status).to eq(200), result.inspect
+
+      opportunities = result.dig('data', 'project', 'ceOpportunities', 'nodes')
+      expect(opportunities.count).to eq(1)
+
+      returned_opportunity = opportunities[0]
+      expect(returned_opportunity['id']).to eq(opportunity.id.to_s)
+    end
+
+    context 'when filtering for open referrals' do
+      let(:variables) do
+        {
+          id: project.id,
+          filters: {
+            status: ['open'],
+          },
+        }
+      end
+
+      let(:opportunity) { create :hmis_ce_opportunity, project: project }
+      let(:closed_opportunity) { create :hmis_ce_opportunity, project: project, status: :closed }
+      let(:locked_opportunity) { create :hmis_ce_opportunity, project: project, status: :locked }
+
+      it 'returns only active referrals' do
+        response, result = post_graphql(**variables) { query }
+        expect(response.status).to eq(200), result.inspect
+
+        opportunities = result.dig('data', 'project', 'ceOpportunities', 'nodes')
+        expect(opportunities.count).to eq(1)
+
+        expect(opportunities).to contain_exactly(
+          a_hash_including('id' => opportunity.id.to_s),
+        )
+      end
+
+      context 'with many opportunities' do
+        before do
+          100.times do
```

Comment:
I would bring this down from 100 just for the sake of test speed, it seems a bit high for testing the n+1. I would think something like 30-40 would suffice (since it's higher than expected db queries `15..20`)